### DISCUSSION TITLE
OpenCL: AMD/ATI GPU 0 (ignored by config): AMD Radeon RX 6600 XT with integrated AMD GPU present

### DISCUSSION THREAD

**Discussion by brierjon** [BOINC/boinc#5217] [Q&A] [answered]
I'm running a [System76 Thelio Mira r3.0](https://tech-docs.system76.com/models/thelio-mira-r3/README.html) with an added RX 6600XT and an AMD 7950X CPU. The system has integrated graphics and discrete graphics as seen in the event log attached as line 6.

I'm trying to figure out two things:

1. Why is the AMD 6600XT GPU on line 5 listed as (ignored by config) which is really not clear why that would be ignored when its the more powerful of the two GPUs. Attaching to projects like Einstein@home result in downloading GPU tasks, then erroring as seen in task 1137282 from Einstein@home. No tasks I've seen attempted to run on the discrete GPU. (possibly related, Pop!_OS is working on [integrated - discrete GPU switching per the Pop!_OS roadmap](https://support.system76.com/articles/roadmap/)).

2. Even so, the assigned GPU tasks to the system try to run with the integrated GPU (gfx1036). No tasks I've seen show the attempt to run on the 6600XT. I'm seeing the same issue of trying to run on integrated GPUs (at least it appears) on other systems on [Milkyway](https://milkyway.cs.rpi.edu/milkyway/results.php?hostid=977138&offset=0&show_names=0&state=6&appid=) and other projects.

I installed BOINC (7.22.0) from the Pop!_OS shop aka their package manager, which it appears I have installed the [Flatpak packaged version](https://flathub.org/apps/edu.berkeley.BOINC). 

**A discovery of possible interest**: Pop!_OS lists two apps named BOINC (the Flatpak, and the apt versions) though the reason they don't display as one entry according to their engineer is the app version ID doesn't match across packaging systems. (a bug ?). If the app IDs matched, there would be one listing with an option of which installation method to use.

I've attached the startup portion of the event log. I haven't been able to track down why the 6600XT is "ignored by config" in any web or local settings.

[eventlog.txt](https://github.com/BOINC/boinc/files/11367227/eventlog.txt)
[einsteinstderr.txt](https://github.com/BOINC/boinc/files/11367282/einsteinstderr.txt)

**Comment by AenBleidd** [comment, marked as answer]
Did you allow multiple GPUs as suggested here https://boinc.berkeley.edu/forum_thread.php?id=13856 ?

**Comment by davidpanderson** [comment]
Does your cc_config.xml contain any <exclude_gpu> or <ignore_ati_dev> elements?
AFAIK this is the only source of "ignored by config".

**Comment by brierjon** [comment]
Thank you for the replies, I wouldn't have assumed that BOINC would harness all GPUs by default or at the very least would assess which GPU is more powerful to be used by default instead of the integrated GPU. It isn't clear from the documentation if you can opt to use just the discrete and not leverage GPU compute on the integrated. 

1. The multiple GPU config is not obvious from the error message. The message usability would improve by stating "ignored by config, if you have multi-gpus they must be enabled in cc_config.xml" with link to the cc_config.xml documentation or ideally just mark use all GPUs in the preferences Options>computing preferences UI.

2. The second issue is still present, the GPU tasks error out regardless of the multi-gpu config (twice as fast with both GPUs trying to compute the task. Pop!_OS currently runs mesa 2.3.5 per their [mesa repo](https://github.com/pop-os/mesa). I'm seeing Flatpak's sandbox says the /usr/ path at system level may be blacklisted, but should be accessible per https://docs.flatpak.org/en/latest/sandbox-permissions.html is there a path issue with the flatpak packaging, mesa, or somewhere else? If flatpak doesn't support GPU compute, wouldn't it be worth stating that in the message for the package and as a notice within the BOINC manager for the user to be aware?  
3. Does the BOINC client/manager perform a basic test for GPGPU support and if that fails, not request GPU tasks? It would seem to save project bandwidth and processing of failed tasks when compute doesn't pass a minimal local test for support on the host system. 

From one task of many (as well as other projects) as an example: [Einstein 1463676856](https://einsteinathome.org/task/1463676856)
`Using OpenCL device "AMD Radeon RX 6600 XT (navi23, LLVM 15.0.7, DRM 3.49, 6.2.6-76060206-generic)" by: AMD
Max allocation limit: 2147483648
Global mem size: 8589934592
OpenCL compiling FAILED! : -11 . Error message: fatal error: cannot open file '/usr/lib/x86_64-linux-gnu/GL/default/share/clc/gfx1032-amdgcn-mesa-mesa3d.bc': No such file or directory

OpenCL device has no FP64 support
OpenCL compiling FAILED! : -11 . Error message: fatal error: cannot open file '/usr/lib/x86_64-linux-gnu/GL/default/share/clc/gfx1032-amdgcn-mesa-mesa3d.bc': No such file or directory

Couldn't compile OpenCL kernels (error: 1)
initialize_ocl returned error [2006]
OCL context null
OCL queue null
Error generating generic FFT context object [5]
23:33:58 (198): [CRITICAL]: ERROR: MAIN() returned with error '5'
FPU status flags: 
mv: cannot stat 'LATeah4021L06_1196.0_0_0.0_6312348_1_0.out': No such file or directory
mv: cannot stat 'LATeah4021L06_1196.0_0_0.0_6312348_1_0.out': No such file or directory
mv: cannot stat 'LATeah4021L06_1196.0_0_0.0_6312348_1_0.out': No such file or directory
mv: cannot stat 'LATeah4021L06_1196.0_0_0.0_6312348_1_0.out': No such file or directory
mv: cannot stat 'LATeah4021L06_1196.0_0_0.0_6312348_1_0.out': No such file or directory
mv: cannot stat 'LATeah4021L06_1196.0_0_0.0_6312348_1_0.out.cohfu': No such file or directory
mv: cannot stat 'LATeah4021L06_1196.0_0_0.0_6312348_1_0.out.cohfu': No such file or directory
mv: cannot stat 'LATeah4021L06_1196.0_0_0.0_6312348_1_0.out.cohfu': No such file or directory
mv: cannot stat 'LATeah4021L06_1196.0_0_0.0_6312348_1_0.out.cohfu': No such file or directory
mv: cannot stat 'LATeah4021L06_1196.0_0_0.0_6312348_1_0.out.cohfu': No such file or directory
mv: cannot stat 'LATeah4021L06_1196.0_0_0.0_6312348_1_0.out.cohfu': No such file or directory
mv: cannot stat 'LATeah4021L06_1196.0_0_0.0_6312348_1_0.out.cohfu': No such file or directory
23:34:09 (198): [normal]: done. calling boinc_finish(69).
23:34:09 (198): called boinc_finish`

**Reply by brierjon** [reply]
Never mind, I read into the state of opencl support on Mesa. There are some statements that the same error looks like it may be an issue with Clover.

As I understand RustCL may bring better Opencl support on Linux coming enabled in an upcoming Mesa if nothing disrupts that. - https://gitlab.freedesktop.org/mesa/mesa/-/merge_requests/15439 - https://www.phoronix.com/news/RadeonSI-Rusticl-Mesa-23.1

For now, I'll have to settle with CPU compute.
